docs/docs/50-user-guide/60-reference-docs/30-promotion-steps/yaml-merge.md

```diff
@@ -0,0 +1,71 @@
+---
+sidebar_label: yaml-merge
+description: Merge multiple YAML file into a single file.
+---
+
+# `yaml-merge`
+
+`yaml-merge` merges multiple YAML files into a single file. This step
+most often used to merge multiple Helm values.yaml files into a single
+file and is commonly followed by a [`helm-template` step](helm-template.md).
+YAML files are merged in order, so the first one is the base, and all
+subsequent files are "overlays", modifying the default values.
```

Comment:
One comment -- two reasons for making it:

`helm-template` step, just like the `helm template` command allows you to specify multiple `values.yaml` files. So:

1. On account of that, merging multiple `values.yaml` files prior to a `helm-template` step probably _won't_ be a particularly common use for this. I do still think the proposed step is _generally_ useful. It would be fine to document what the step does without calling out any specific, common use.

1. I suspect your own use case actually varies from the above, but on the odd chance that it does not, I just wanted you to know about existing options already available to you.

Reply:
I also explained my use case in another comment. Not sure if you want that use-case to appear in the docs ?

Reply:
> Your use case is a bit complex. (This is not to say invalid. It just takes some effort to understand.) I'm not positive we'd be doing readers any favors by asking them to understand that as a means to seeing the utility of this step. i.e. Don't want to demo simple functionality with a complex use case.

I updated the docs to simply explain how it work and not describe my use case. I'm not sure readers needs a complex use case here. Basic use case is simple: you have many yaml files and you want to merge them into one.

The description of the use case for reviewers of the PR/issue is at https://github.com/akuity/kargo/pull/4634#discussion_r2316155956

Please let me know if you feel better with this example.